x/reward/types/reward_pool.go

```diff
@@ -12,17 +12,29 @@ func NewRewardPool(launchID uint64, currentRewardHeight int64) RewardPool {
 	return RewardPool{
 		LaunchID:            launchID,
 		CurrentRewardHeight: uint64(currentRewardHeight),
+		Closed:              false,
 	}
 }
 
 // Validate check the RewardPool object
 func (m RewardPool) Validate() error {
-	if m.Coins.Empty() {
+	if m.InitialCoins.Empty() {
 		return errors.New("empty reward pool coins")
 	}
-	if err := m.Coins.Validate(); err != nil {
+	if err := m.InitialCoins.Validate(); err != nil {
 		return fmt.Errorf("invalid reward pool coins: %s", err)
 	}
+	if m.CurrentCoins.Empty() {
+		return errors.New("empty reward pool coins")
+	}
+	if err := m.CurrentCoins.Validate(); err != nil {
+		return fmt.Errorf("invalid reward pool coins: %s", err)
+	}
+
+	if m.CurrentCoins.IsAnyGTE(m.InitialCoins) {
```

Comment:
We should ensure that you can't have a denom in `CurrentCoins` not present in `InitialCoins`, does this check verify this? We should add a unit test for this case